music_assistant/providers/musiccast/__init__.py

```diff
@@ -0,0 +1,790 @@
+        ):
+            if queue.current_item is not None:
+                if update_helper.current_queue_item_id == queue.current_item.queue_item_id:
+                    player.current_media = queue.current_item  # type: ignore[assignment]
+            if queue.next_item is not None:
+                if update_helper.current_queue_item_id == queue.next_item.queue_item_id:
+                    player.current_media = queue.next_item  # type: ignore[assignment]
+        else:
+            player.current_media = PlayerMedia(
+                uri=f"{player.player_id}_{device.source_id}",
+                title=device.media_title,
+                artist=device.media_artist,
+                album=device.media_album_name,
+                image_url=device.media_image_url,
+            )
+
+        # SOURCE
+        if device.source_id == "server" and update_helper.controlled_by_mass:
+            player.active_source = queue.queue_id if queue is not None else None
+        else:
+            player.active_source = device.source_id
+
+        # GROUPING
+        # A zone cannot be synced to another zone or main of the same device.
+        # Additionally, a zone can only be synced, if main is currently not using any netusb
+        # function.
+        # For a Zone which will be synced to main, grouping emits a "main_sync" instead
+        # of a mc link.
+
+        # TODO: Revisit this - it produces the expected output (?), but the ui
+        # does not update correctly?
+        can_group_with = set(self.mc_controller.all_zone_devices)
+        if len(device.physical_device.zone_devices) > 1:
+            # receiver with zones
+            _main_zone = device.physical_device.zone_devices.get("main")
+            if _main_zone is None:
+                return
+            _other_zones = set(_main_zone.other_zones)
+
+            if device == _main_zone:
+                # a main zone cannot join another zone, but a zone can join
+                # main, see below.
+                can_group_with.difference_update(_other_zones)
+            elif device in _other_zones:
+                # can_group_with.difference_update(_other_zones.difference({device}))
+                # enforce a zone to be either the server, or sync to main of
+                # same device. makes life much easier.
+                can_group_with = {device, _main_zone}
+
+        # player.can_group_with = {
+        #     x.ma_player_id for x in can_group_with if x.ma_player_id is not None
+        # }
+        player.can_group_with = {self.instance_id}
+
+        if len(device.musiccast_group) == 1:
+            if device.musiccast_group[0] == device:
+                # we are in a group with ourselves.
+                player.group_childs = UniqueList([])
+                player.synced_to = None
+                player.active_group = None
+
+        elif not device.is_client and not device.is_server:
+            player.group_childs = UniqueList([])
```

Comment:
I'm pretty sure we have a clear method on the list so you dont have to assign yourself